agent/secrets/secrets.go

```diff
@@ -60,15 +60,20 @@ func (s *secrets) Reset() {
 
 // taskRestrictedSecretsProvider restricts the ids to the task.
 type taskRestrictedSecretsProvider struct {
-	secrets   exec.SecretGetter
-	secretIDs map[string]struct{} // allow list of secret ids
+	secrets        exec.SecretGetter
+	secretIDs      map[string]struct{} // allow list of secret ids
+	secretMappings map[string]string   // for mapping referenced secret id to an internal secret
```

Comment:
is the information in `secretMappings` not a superset of the information in `secretIDs`? or does `secretIDs` also include the ephemeral task-specific secrets?

Reply:
The information in `secretMappings` only overlaps with `secretIDs`, as in all the keys of `secretMappings` are present in `secretIDs`, but not all the keys of `secretIDs` are necessarily present in `secretMappings`. Besides, the values in `secretMappings` are only present in that map, as `secretIDs` is a `map[string]struct{}`.